pkg/host/host.go

```diff
@@ -0,0 +1,455 @@
+		return err
+	}
+	return nil
+}
+
+func (h *HostManager) isKernelModuleLoaded(kernelModuleName string) (bool, error) {
+	glog.Infof("isKernelModuleLoaded(): check if kernel module %s is loaded", kernelModuleName)
+	chrootDefinition := getChrootExtention(h.RunOnHost)
+
+	stdout, stderr, err := h.cmd.Run("/bin/sh", "-c", fmt.Sprintf("%s lsmod | grep \"^%s\"", chrootDefinition, kernelModuleName))
+	if err != nil && stderr.Len() != 0 {
+		glog.Errorf("isKernelModuleLoaded(): failed to check if kernel module %s is loaded: error: %v stderr %s", kernelModuleName, err, stderr.String())
+		return false, err
+	}
+	glog.V(2).Infof("isKernelModuleLoaded(): %v", stdout.String())
+	if stderr.Len() != 0 {
+		glog.Errorf("isKernelModuleLoaded(): failed to check if kernel module %s is loaded: error: %v stderr %s", kernelModuleName, err, stderr.String())
+		return false, fmt.Errorf(stderr.String())
+	}
+
+	if stdout.Len() != 0 {
+		glog.Infof("isKernelModuleLoaded(): kernel module %s already loaded", kernelModuleName)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (h *HostManager) TryEnableTun() {
+	if err := h.LoadKernelModule("tun"); err != nil {
+		glog.Errorf("tryEnableTun(): TUN kernel module not loaded: %v", err)
+	}
+}
+
+func (h *HostManager) TryEnableVhostNet() {
+	if err := h.LoadKernelModule("vhost_net"); err != nil {
+		glog.Errorf("tryEnableVhostNet(): VHOST_NET kernel module not loaded: %v", err)
+	}
+}
+
+func (h *HostManager) TryEnableRdma() (bool, error) {
+	glog.V(2).Infof("tryEnableRdma()")
+	chrootDefinition := getChrootExtention(h.RunOnHost)
+
+	// check if the driver is already loaded in to the system
+	_, stderr, mlx4Err := h.cmd.Run("/bin/sh", "-c", fmt.Sprintf("grep --quiet 'mlx4_en' <(%s lsmod)", chrootDefinition))
+	if mlx4Err != nil && stderr.Len() != 0 {
+		glog.Errorf("tryEnableRdma(): failed to check for kernel module 'mlx4_en': error: %v stderr %s", mlx4Err, stderr.String())
+		return false, fmt.Errorf(stderr.String())
+	}
+
+	_, stderr, mlx5Err := h.cmd.Run("/bin/sh", "-c", fmt.Sprintf("grep --quiet 'mlx5_core' <(%s lsmod)", chrootDefinition))
+	if mlx5Err != nil && stderr.Len() != 0 {
+		glog.Errorf("tryEnableRdma(): failed to check for kernel module 'mlx5_core': error: %v stderr %s", mlx5Err, stderr.String())
+		return false, fmt.Errorf(stderr.String())
+	}
+
+	if mlx4Err != nil && mlx5Err != nil {
+		glog.Errorf("tryEnableRdma(): no RDMA capable devices")
+		return false, nil
+	}
+
+	isRhelSystem, err := h.isRHELSystem()
+	if err != nil {
+		glog.Errorf("tryEnableRdma(): failed to check if the machine is base on RHEL: %v", err)
+		return false, err
+	}
+
+	// RHEL check
+	if isRhelSystem {
+		return h.enableRDMAOnRHELMachine()
+	}
+
+	isUbuntuSystem, err := h.isUbuntuSystem()
+	if err != nil {
+		glog.Errorf("tryEnableRdma(): failed to check if the machine is base on Ubuntu: %v", err)
+		return false, err
+	}
+
+	if isUbuntuSystem {
+		return h.enableRDMAOnUbuntuMachine()
+	}
+
+	osName, err := h.getOSPrettyName()
```

Comment:
line 167 returns result if isRhelSystem
link 176 returns result if isUbuntuSystem

what other OSs are supported to get into this code?

Reply:
@SchSeba ^^